tools/loader/ImageClassifier.cpp

```diff
@@ -267,20 +303,22 @@ static void processAndPrintResultsImpl(Tensor *SMT,
 
     auto topKPairs = getTopKPairs(SH);
     printTopKPairs(topKPairs);
+    retVal += checkExpectedLabel(topKPairs, fileName,
```

Comment:
Since this call basically just returns 0 immediately if `expectedMatchingLabels.empty()`, perhaps you could just replace this by:

```
if (!expectedMatchingLabels.empty()) {
  retVal += checkExpectedLabel(topKPairs, fileName, expectedMatchingLabels[i]);
}
```

Then you could also remove the "default value of category" check inside `checkExpectedLabel()`.